core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java

```diff
@@ -179,6 +179,7 @@ public final synchronized RunT newBuild() throws IOException {
         } catch (InstantiationException e) {
             throw new Error(e);
         } catch (IllegalAccessException e) {
+            LOGGER.log(Level.WARNING, String.format("A new build could not be created in job %s", asJob().getFullName()), e);
```

Comment:
Btw, looking at that code again, I'm wondering if the fact there are many exceptions catched isn't just coming from the fact it was to catch all the "reflective operations exceptions", in pre Java 7.

`git blame` seems to tell me @jglick wrote that code. Maybe he could confirm, but I suppose that could could possibly be simplified using the new Java 7's introduced exception `ReflectiveOperationException` as:

```java
try {
    RunT lastBuild = getBuildClass().getConstructor(asJob().getClass()).newInstance(asJob());
    builds.put(lastBuild);
    lastBuild.getPreviousBuild(); // JENKINS-20662: create connection to previous build
    return lastBuild;
} catch (InvocationTargetException e) {
    throw handleInvocationTargetException(e);
} catch (ReflectiveOperationException e) {
    LOGGER.log(Level.WARNING, String.format("A new build could not be created in job %s", asJob().getFullName()), e);
    throw new Error(e);
}
```